registries/_format/sf-integer.md

```diff
@@ -4,7 +4,7 @@ issue:
 description: structured fields integer as defined in [RFC8941]
 source: https://www.rfc-editor.org/rfc/rfc8941#name-integers
 source_label: RFC 8941
-base_type: integer, number
+base_type: [integer, number]
```

Comment:
Based on what we use more it looks like this is indeed the correct format. You can check also [decimal](https://github.com/OAI/OpenAPI-Specification/blob/gh-pages/registries/_format/decimal.md) and [decimal128](https://github.com/OAI/OpenAPI-Specification/blob/gh-pages/registries/_format/decimal128.md).

One more note, I am not so sure that the representations in these files are THAT important. The reason is that these files seem to just be for building the page and not for documentation. Which means that we mostly care for the end result and how the data will appear when rendered.

Reply:
I agree with this:
> we mostly care for the end result and how the data will appear when rendered.

and thanks for the link to the `decimal` format because that shows how problematic the change actually is.

<img width="759" alt="image" src="https://github.com/OAI/OpenAPI-Specification/assets/85643503/97fe6b94-8ef5-4cd6-9cc3-ad275a493ce4">

I would argue that this is _not_ how we want the data to appear, so the array format just looks wrong to me.

If anything, I think we should be going the other direction, to make all these just a comma-separated string.

Reply:
@mikekistler That's what the PR fixes. They will appear as `string, number`.

Reply:
@Bellangelo I think what @mikekistler is getting at is that we just need the existing `string, number` to be rendered as `string, numer`, rather than rendering it as something that looks like a tuple with first a string and then a number.

Reply:
Hi @handrews and @mikekistler, I think there is a misunderstanding here, so I will try to explain the situation the best I can.
So, the line that @mikekistler commented on is a variable for the [Liquid](https://shopify.github.io/liquid/) template engine that Jekyll uses. The reason why the `sf-integer` and the `decimal` appear different is that the `sf-integer` sets the variable type to a string, while the other to an array. Liquid by default join array's item together without any separator. This is solved by this [line](https://github.com/OAI/OpenAPI-Specification/pull/3789/files#diff-5d645f1c35572babebe7fa6962b64d2f54ef346f51c28166e77b2e8aee3bb2ffR27) which overrides the default behavior and instead uses the separator that we want, a comma with a space.
Did you test it and had different results? 

While we could easily just use the string instead, so keeping the `sf-integer` as is and changing the rest, the reason I persist on this is that it feels kind of weird to me that the produced JSON would have a `base_type` which is a `string` but it contains actually 2 values.
From this point, there are 2 ways to go to:
1. **You agree with the point i made on the JSON part**:  Then if we keep the variable type as a string we would need to create a more complex filter as we would need to split the string and generate a valid JSON array.
2. **You don't agree with the point I made on the JSON part**:  Then I revert the changes on how we produce the JSON and I change every format that uses an array.
3. Of course, there is a secret 3rd option which you can just suggest something that i missed.

Reply:
@Bellangelo Ah, yes I see, thank you for clarifying!  I agree that sf-integer should be an array here, (matching how it is in decimal, for example).  I got confused thinking that all of the others did _not_ wrap that list in an array.  But they do.  My apologies!  I read through this too quickly.

@mikekistler I think this is correct, or at least it's the same as the other formats that use either of two types.  If this isn't the way we want to structure type alternative lists, we should probably sort that out separately from this bugfix.

Reply:
Thanks for the extended explanation @Bellangelo! This looks good then.